src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -0,0 +1,364 @@
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsExtractLabelsAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsExtractLabelsAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterExtent(
+                  QStringLiteral( "EXTENT" ),
+                  QObject::tr( "Map extent" ) ) );
+
+  addParameter( new QgsProcessingParameterScale(
+                  QStringLiteral( "SCALE" ),
+                  QObject::tr( "Map scale" ) ) );
+
+  addParameter( new QgsProcessingParameterNumber(
+                  QStringLiteral( "DPI" ),
+                  QObject::tr( "Map resolution (in DPI)" ),
+                  QgsProcessingParameterNumber::Double,
+                  QVariant( 96.0 ) ) );
+
+  addParameter( new QgsProcessingParameterMapTheme(
+                  QStringLiteral( "MAP_THEME" ),
+                  QObject::tr( "Map theme" ),
+                  QVariant(), true ) );
+
+  addParameter( new QgsProcessingParameterFeatureSink(
+                  QStringLiteral( "OUTPUT" ),
+                  QObject::tr( "Extracted labels" ),
+                  QgsProcessing::TypeVectorPoint ) );
+}
+
+QString QgsExtractLabelsAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Extract label information (position, font details, etc.) from a rendered map at a given extent and scale." );
+}
+
+QString QgsExtractLabelsAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm extracts label information from a rendered map at a given extent and scale.\n\n"
+                      "If a map theme is provided, the rendered map will match the visibility and symbology of that theme. If left blank, the project's layer tree setup will be used.\n\n"
+                      "Extracted label information include position, font family, size, weight, rotation, and more." );
+}
+
+QgsExtractLabelsAlgorithm *QgsExtractLabelsAlgorithm::createInstance() const
+{
+  return new QgsExtractLabelsAlgorithm();
+}
+
+class ExtractLabelSink : public QgsLabelSink
+{
+  public:
+    ExtractLabelSink( QMap<QString, QString> *mapLayerNames, QgsProcessingFeedback *feedback )
+      : mMapLayerNames( mapLayerNames )
+      , mFeedback( feedback )
+    {
+    };
+
+    void drawLabel( const QString &layerId, QgsRenderContext &context, pal::LabelPosition *label, const QgsPalLayerSettings &settings ) override
+    {
+      const QgsFeatureId fid = label->getFeaturePart()->featureId();
+      if ( settings.placement == QgsPalLayerSettings::Curved ||
+           settings.placement == QgsPalLayerSettings::PerimeterCurved )
+      {
+        mFeedback->pushWarning( QObject::tr( "Curved placement not supported, skipping label for FID %1 from layer %2" ).arg( fid ).arg( layerId ) );
+        return;
+      }
+
+      QgsTextLabelFeature *labelFeature = dynamic_cast<QgsTextLabelFeature *>( label->getFeaturePart()->feature() );
+      if ( !labelFeature )
+        return;
```

Comment:
Worth a message?

Reply:
Actually, after looking at other parts of our code, this condition can be removed safely.